### DISCUSSION TITLE
Splitting multiple choice values

### DISCUSSION THREAD

**Discussion by 0204xls** [microsoft-search/pnp-modern-search#3185] [Q&A]
I am having trouble with multiple selection choice columns in Modern Search. As other posts have suggested, refinable properties are concatenated into a string containing all selected choice values separated by ;#. May anyone suggest a way to break these up strings up in the search results so that an item can be filtered by its individual constituent values? For instance, if one item had the selections Choice1, Choice2, and Choice3, they could filter by any one of those values on their own rather than one string with the value "Choice1;#Choice2;#Choice3;#"? See image below for an example.

![sampleMultiValue](https://github.com/microsoft-search/pnp-modern-search/assets/137092197/3b9619f3-1624-49cb-8bc8-25cea72e4843)

In this example, the values are from a choice column with multiple selections enabled mapped to a RefinableStringXX property.

I would greatly appreciate any advice on the matter. Thanks in advance.

**Comment by wobba** [comment]
Here's a sample, and replace the inline string there with something like `item.ManagedPropertyName`.

```html
{{#each (split "foo;#bar;#foobar" ";#") as |value|}}
            <p>{{value}}</p>
{{/each}}
```

**Reply by 0204xls** [reply]
Thank you for this. Can I place this somewhere in the filter template? I achieved some success in the results template, but that only altered how the results appeared, not the filter functionality.

**Reply by wobba** [reply]
Try a custom Filter template and add it. You likely have to test a bit and massage what the value should be as you are splitting the values to be passed down as well.
